.github/workflows/changelog.yml

```diff
@@ -0,0 +1,33 @@
+# This action requires that any PR targeting the main branch should touch at
+# least one CHANGELOG file. If a CHANGELOG entry is not required, add the "Skip
+# Changelog" label to disable this action.
+
+name: changelog
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+    branches:
+      - main
+jobs:
+  changelog:
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.pull_request.labels.*.name, 'Skip Changelog')"
+
+    steps:
+      - uses: actions/checkout@v2
```

Comment:
You can add `fetch-depth: 1` instead of manually running, similar to

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/.github/workflows/ci.yml#L15